src/Acuminator/Acuminator.Tests/Tests/StaticAnalysis/DacKeyFieldDeclaration/KeyFieldDacTests.cs

```diff
@@ -0,0 +1,141 @@
+﻿using Acuminator.Analyzers.StaticAnalysis;
+using Acuminator.Analyzers.StaticAnalysis.DacExtensionDefaultAttribute;
+using Acuminator.Analyzers.StaticAnalysis.DacKeyFieldDeclaration;
+using Acuminator.Tests.Helpers;
+using Acuminator.Tests.Verification;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Acuminator.Tests.Tests.StaticAnalysis.DacKeyFieldDeclaration
+{
+	public class KeyFieldInDacTests : CodeFixVerifier
+	{
+		protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer() => new KeyFieldDeclarationAnalyzer();
+
+		protected override CodeFixProvider GetCSharpCodeFixProvider() => new KeyFieldDeclarationFix();
+
+		[Theory]
+		[EmbeddedFileData("CompoundKey.cs")]
+		public async Task TestCompoundKey_ShouldNotShowDiagnosticAsync(string source) =>
+			await VerifyCSharpDiagnosticAsync(source);
+
+		[Theory]
+		[EmbeddedFileData("Key.cs")]
+		public async Task TestKey_ShouldNotShowDiagnosticAsync(string source) =>
+			await VerifyCSharpDiagnosticAsync(source);
+
+		[Theory]
+		[EmbeddedFileData("IdentityKey.cs")]
+		public async Task TestIdentityKey_ShouldNotShowDiagnosticAsync(string source) =>
+			await VerifyCSharpDiagnosticAsync(source);
+
+		[Theory]
+		[EmbeddedFileData("IdentityKey_Key.cs")]
+		public async Task TestIdentityKeyWithKeyAsync(string source) =>
+			await VerifyCSharpDiagnosticAsync(source,
+				Descriptors.PX1055_DacKeyFieldsWithIdentityKeyField.CreateFor(
+					locations: new (int Line, int Column)[] 
+					{
+						(Line: 10, Column: 4),
```

Comment:
@saloponov why are there multiple diagnostics for the same location?

Reply:
@dropsonic Because location contains next elements: {`location`, `extraLications`}, where `extraLocation` contains `location`. Analyzer don't exclude current `location` from `extraLocations`. (Possible `extraLocations` have different name (`locations`)).